### PR TITLE
[CodeWhisperer] add cwspr support of jsx file

### PR DESCRIPTION
**Pull request by Will-ShaoHua**
corresponding PR for JB: https://github.com/aws/aws-toolkit-vscode/pull/2859

https://user-images.githubusercontent.com/96078566/187651960-ceb195a9-2b48-4a93-a6dc-5502a736fab3.mov

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [ ] I have read the **README** document
- [ ] I have read the **CONTRIBUTING** document
- [ ] Local run of `gradlew check` succeeds
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if change is customer facing in the IDE.
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/editor/CodeWhispererEditorUtil.kt [hunk 1]
```diff
@@ -36,24 +37,22 @@ object CodeWhispererEditorUtil {
...
-
     val PsiFile.programmingLanguage: ProgrammingLanguage
-        get() = ProgrammingLanguage(lowerCase(this.fileType.name))
+        get() {
+            val fileTypeName = lowerCase(this.fileType.name)
+            return when {
+                fileTypeName.contains("jsx") -> ProgrammingLanguage(CodewhispererLanguage.Javascript)
+                else -> ProgrammingLanguage(fileTypeName)
+            }
+        }
```
Comment (andrewyuq) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/editor/CodeWhispererEditorUtil.kt:47]: Do we want to add a new value for jsx in telemetry?
For the final mapping when we are creating the request we can map `jsx harmony` to `javascript`, but for other purposes (e.g. sending telemetry events) we can get events of .jsx files if we want.

Comment (andrewyuq) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/editor/CodeWhispererEditorUtil.kt:47]: Also -- consider moving the mapping helper into CodeWhispererLanguageManager -- looks cleaner IMO.

Comment (Will-ShaoHua) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/editor/CodeWhispererEditorUtil.kt]: > Do we want to add a new value for jsx in telemetry? For the final mapping when we are creating the request we can map `jsx harmony` to `javascript`, but for other purposes (e.g. sending telemetry events) we can get events of .jsx files if we want.


Yea sounds good, might need another type of enum CodeWhispererLanguage to represent JSX. Will open another PRs update definition + plugin telemetry sending.

Comment (Will-ShaoHua) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/editor/CodeWhispererEditorUtil.kt]: > Also -- consider moving the mapping helper into CodeWhispererLanguageManager -- looks cleaner IMO.

updated!

**Pull request closed without merging**